crates/holochain_state/src/query.rs

```diff
@@ -308,16 +308,21 @@ impl<'stmt> Store for Txn<'stmt, '_> {
         }
     }
 
+    // Returns only actions that are referenced by at least one integrated dht op.
     fn get_action(&self, hash: &ActionHash) -> StateQueryResult<Option<SignedActionHashed>> {
-        let shh = self.txn.query_row(
+        let action = self.txn.query_row(
             "
             SELECT
             Action.blob, Action.hash
             FROM Action
-            WHERE hash = :hash
+            JOIN DhtOp ON Action.hash = DhtOp.action_hash
+            WHERE Action.hash = :hash
+            AND DhtOp.validation_status = :validation_status
+            LIMIT 1
```

Comment:
Why limit 1? If there is more than one matching hash then that is an error in the data integrity and the query should fail I think